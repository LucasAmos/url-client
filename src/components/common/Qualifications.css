.qualifications {
  margin: 70px 0px 70px 0px;
  margin-bottom: 50px;
  font-family: 'Roboto';
}

.qualifications h2 {
  font-size: 26px;
  padding-top: 10px;
  padding-bottom: 20px;
}

.qualifications p {
  font-size: 18px;
  font-family: 'Roboto';
  font-weight: 100;
  text-align: justify;
}

.qualifications .aston, .qualifications .st-andrews {
  text-align: center;
}

.qualifications .standrews-row .portfolio-image {
  height: 200px;
  border: 8px solid rgba(211, 211, 211, 0.2);
  margin-right: 20px;
  transition: all .2s ease-in-out;
}

.qualifications .standrews-row .portfolio-image:hover {
  transform: scale(1.1);
}

.qualifications .gaudeamus {
  display: block;
  background-repeat: no-repeat;
  background-size: contain;
  background-position: center bottom;
  height: 400px;
  background: url('../../res/images/standrews.jpg') no-repeat center center scroll;
  background-size: cover;
}

.qualifications .campus {
  display: block;
  background-repeat: no-repeat;
  background-size: contain;
  background-position: center bottom;
  height: 400px;
  background: url('../../res/images/aston.jpeg') no-repeat center center scroll;
  background-size: cover;
}

.qualifications .standrews-row {
margin-bottom: 70px;
}

.qualifications .standrews-row h3{
  font-family: 'Roboto';
  font-weight: 500;
  font-size: 16px;
  margin-bottom:10px;
}

.qualifications .title {
  text-align: center;
  font-family: 'Roboto Condensed';
}

.qualifications .title h2 {
  font-size: 40px;
  margin-bottom: 0px;
  padding-bottom: 0px;
}

.qualifications .title h3 {
  font-size: 24px;
  margin-bottom: 40px;
}

.qualifications .github-logo {
  width: 35px;
  height: 35px;
  border: 0px;
  vertical-align: bottom;
  transition: all .2s ease-in-out;
}

.qualifications .github-logo:hover {
  transform: scale(1.1);
}

@media (max-width: 1199px){
.qualifications .my-work {
  display: none;
  }
}

@media (max-width: 400px){
  .qualifications .title h3 {
    font-size: 20px;
  }
  }

@media (max-width: 500px) {
  .qualifications  {
    margin-top: 30px;
    margin-bottom: 40px ;
  }

  .qualifications .uni-row {
    margin-bottom: 20px;
  }

  .qualifications .aston-row{
    margin-top: 50px;
  }

  .qualifications h2 {
    padding-bottom: 0px;
  }
}

@media (max-width: 991px){

  .qualifications #github {
    display: none;
  }
}

.qualifications #github-narrow {
  display: none;
}

@media (min-width: 992px) and (max-width: 1199px){
  .qualifications #github-narrow {
    display: block;
  }
}
