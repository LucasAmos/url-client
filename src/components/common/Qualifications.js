/* eslint-disable react/jsx-one-expression-per-line */
import React from 'react';
import './Qualifications.css';
import { Container, Row, Col } from 'react-bootstrap';
import Fade from 'react-reveal/Fade';
import textual from '../../res/images/textual.png';
import masters from '../../res/images/masters.png';
import github from '../../res/images/github.png';

export default function Qualifications() {
  return (
    <div className="qualifications">
      <Container>
        <Row className="title">
          <Col>
            <Fade duration={1500}>
              <h2>
                HERE&apos;S SOME OF MY EDUCATION
              </h2>
              <h3>
                (I LIKE TO LEARN THINGS)
              </h3>
            </Fade>
          </Col>
        </Row>
        <Row className="standrews-row uni-row">
          <Col
            xs={{ order: 2, span: 12 }}
            lg={{ span: 6, offset: 0 }}
          >
            <Fade>
              <div className="gaudeamus" />
            </Fade>
          </Col>
          <Col
            xs={{ order: 1, span: 12 }}
            lg={{ span: 5, offset: 0 }}
          >
            <Fade>
              <h2>UNIVERSITY OF ST ANDREWS</h2>

              <p>
                I hold a Masters degree in Advanced Computer Science from the
                University of St Andrews where I competed for the university
                in national archery competitions.
              </p>

              <div className="my-work">
                <h3> See some of my work</h3>
                <a
                  href="https://storage.googleapis.com/url-images-22/Alternatives%20to%20textual%20authentication.pdf"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <img
                    className="portfolio-image"
                    src={textual}
                    alt="alternatives to textual authentication"
                  />
                </a>
                <a
                  href="https://storage.googleapis.com/url-images-22/standrewsposter.pdf"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <img
                    className="portfolio-image"
                    src={masters}
                    alt="masters poster"
                  />
                </a>
                <a
                  href="https://github.com/LucasAmos/Masters-project"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <img
                    src={github}
                    className="github-logo"
                    alt="github logo"
                  />
                </a>
              </div>
            </Fade>
          </Col>
        </Row>
        <Row className="aston-row uni-row">
          <Col
            xs={{ order: 2, span: 12 }}
            lg={{ span: 6, order: 1 }}
            className="aston"
          >
            <Fade>
              <div className="campus" />
            </Fade>
          </Col>
          <Col
            xs={{ order: 1, span: 12 }}
            lg={{ span: 5, order: 2 }}
            className="bsc"
          >
            <Fade cascade>
              <div>
                <h2>ASTON UNIVERSITY</h2>
                <p> I also hold a first class BSc in Computer Science from Aston University.</p>
                <p>
                  While studying for my undergraduate degree I was the President of
                  Aston Computer Science Society, the university&apos;s largest academic society.
                </p>
                <p>
                  I particpated in the UK finals of the CapGemini SuperTechies business IT
                  competition, finishing a close runner up in second place.
                </p>
                <a
                  href="https://github.com/LucasAmos/Undergraduate-project"
                  target="_blank"
                  rel="noopener noreferrer"
                >
                  <img
                    src={github}
                    id="github"
                    className="github-logo"
                    alt="github logo"
                  />
                </a>
              </div>
            </Fade>
          </Col>
        </Row>
      </Container>
    </div>
  );
}
