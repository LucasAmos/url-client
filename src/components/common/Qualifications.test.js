import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Qualifications from './Qualifications';

describe('Portfolio component', () => {
  let savedError;
  beforeEach(() => {
    savedError = console.error;
    console.error = jest.fn();
  });

  afterEach(() => {
    console.error = savedError;
  });

  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      shallow(
        <Qualifications />,
      ),
      div,
    );
    ReactDOM.unmountComponentAtNode(div);
  });
  it('matches the snapshot', () => {
    const tree = renderer.create(
      <Qualifications />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
