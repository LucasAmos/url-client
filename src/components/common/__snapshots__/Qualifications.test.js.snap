// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Portfolio component matches the snapshot 1`] = `
<div
  className="qualifications"
>
  <div
    className="container"
  >
    <div
      className="title row"
    >
      <div
        className="col"
      >
        <h2
          className="react-reveal"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          HERE'S SOME OF MY EDUCATION
        </h2>
        <h3
          className="react-reveal"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          (I LIKE TO LEARN THINGS)
        </h3>
      </div>
    </div>
    <div
      className="standrews-row uni-row row"
    >
      <div
        className="col-lg-6 col-12 offset-lg-0 order-2"
      >
        <div
          className="react-reveal gaudeamus"
          style={
            Object {
              "opacity": undefined,
            }
          }
        />
      </div>
      <div
        className="col-lg-5 col-12 offset-lg-0 order-1"
      >
        <h2
          className="react-reveal"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          UNIVERSITY OF ST ANDREWS
        </h2>
        <p
          className="react-reveal"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          I hold a Masters degree in Advanced Computer Science from the University of St Andrews where I competed for the university in national archery competitions.
        </p>
        <div
          className="react-reveal my-work"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          <h3>
             See some of my work
          </h3>
          <a
            href="https://storage.googleapis.com/url-images-22/Alternatives%20to%20textual%20authentication.pdf"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="alternatives to textual authentication"
              className="portfolio-image"
              src="textual.png"
            />
          </a>
          <a
            href="https://storage.googleapis.com/url-images-22/standrewsposter.pdf"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="masters poster"
              className="portfolio-image"
              src="masters.png"
            />
          </a>
          <a
            href="https://github.com/LucasAmos/Masters-project"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="github logo"
              className="github-logo"
              src="github.png"
            />
          </a>
        </div>
      </div>
    </div>
    <div
      className="aston-row uni-row row"
    >
      <div
        className="aston col-lg-6 col-12 order-lg-1 order-2"
      >
        <div
          className="react-reveal campus"
          style={
            Object {
              "opacity": undefined,
            }
          }
        />
      </div>
      <div
        className="bsc col-lg-5 col-12 order-lg-2 order-1"
      >
        <div
          className="react-reveal"
          style={
            Object {
              "opacity": undefined,
            }
          }
        >
          <h2>
            ASTON UNIVERSITY
          </h2>
          <p>
             I also hold a first class BSc in Computer Science from Aston University.
          </p>
          <p>
            While studying for my undergraduate degree I was the President of Aston Computer Science Society, the university's largest academic society.
          </p>
          <p>
            I particpated in the UK finals of the CapGemini SuperTechies business IT competition, finishing a close runner up in second place.
          </p>
          <a
            href="https://github.com/LucasAmos/Undergraduate-project"
            rel="noopener noreferrer"
            target="_blank"
          >
            <img
              alt="github logo"
              className="github-logo"
              id="github"
              src="github.png"
            />
          </a>
        </div>
      </div>
    </div>
  </div>
</div>
`;
